pyproject.toml

```diff
@@ -1,7 +1,42 @@
-[build-system]
-requires = ["setuptools", "setuptools-git-versioning", "wheel"]
-build-backend = "setuptools.build_meta"
+[tool.poetry]
+name = "vss-tools"
+homepage="https://github.com/COVESA/vss-tools"
+description='COVESA Vehicle Signal Specification tooling.'
+authors = ["COVESA VSS <covesa-dev+subscribe@covesa.global>"]
+version = "4.2.0"
```

Comment:
Does poetry support automatic versioning based on git tags. We quite recently changed vss-tools so it tries to take version from git tag if present and also including it in pypi builds. Here we go back to hard-coded version, right?

Reply:
No idea but I would go the other way around to be honest.
I had quite some good experience with [bump2version](https://pypi.org/project/bump2version/).

With that you can specify the version only once in the root of the project and define other files where it is used to be bumped automatically. So if we have the version only present in the pyproject.toml the .bumpversion.cfg would look like this:

```
[bumpversion]
current_version = 4.2.0

[bumpversion:file:pyproject.toml]
```

We can then use `bump2version --allow-dirty minor` to bump the version to `4.3.0`.

In our master workflow we can then just call our tool with `vspec2json --version` parse the output and tag the commit accordingly.

Reply:
Or with .bumpversion.cfg in place we could directly mention the tag in the workflow file aswell to bump it

Reply:
I did some googling and there seems to be some attempts to use tag as source like https://github.com/tiangolo/poetry-version-plugin, but it seems to expect that last commit is tagged.

I also saw [this article](https://nadeauinnovations.com/post/2020/08/one-version-to-rule-them-all-keeping-your-python-package-version-number-in-sync-with-git-and-poetry/) where they give an example on how to update version based on git tag before building.

```
poetry version $(git describe --tags --abbrev=0)
poetry build
```

To give some background, in the past we relied on hard-coded versions. After we released a version (lets say `5.4`) we changed version on main-branch to something like `5.5.dev0` so that if anyone built/run from main it would be obvious that they did not use the official `5.4` version. With the current "automated" version based on tags it would instead report something like `5.4.post2.dirty-3478349834` . I see no real problem in going back to having a version hard-coded in `pyproject.toml`.

Reply:
And as of today for vss-tools we have only version in one place, so updating it manually with "poetry version X.Y.Z" would work fine. After this PR is merged when preparing next release we would anyhow need to update the (informal) release instructions at https://github.com/COVESA/vehicle_signal_specification/wiki/Release-Instructions-and-Checklist